presto-main/src/main/java/com/facebook/presto/util/GraphvizPrinter.java

```diff
@@ -129,18 +133,24 @@
         Preconditions.checkState(NODE_COLORS.size() == NodeType.values().length);
     }
 
-    private GraphvizPrinter() {}
+    private final ConnectorSession session;
 
-    public static String printLogical(List<PlanFragment> fragments)
+    private GraphvizPrinter(Session session)
     {
+        this.session = requireNonNull(session).toConnectorSession();
+    }
+
+    public static String printLogical(List<PlanFragment> fragments, Session session)
+    {
+        GraphvizPrinter printer = new GraphvizPrinter(session);
```

Comment:
no use for printer. Session can be a local variable. No need for putting in the constructor.